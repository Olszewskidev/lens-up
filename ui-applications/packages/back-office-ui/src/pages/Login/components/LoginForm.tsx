import { FormComponent, SubmitFormButtonComponent } from "@lens-up/shared-components";

interface ILoginFormProps {
    handleFormSubmit: () => void,
}

const LoginForm = ({ handleFormSubmit }: ILoginFormProps) => {
    return (
        <FormComponent handleFormSubmit={handleFormSubmit}>
            <div className="flex items-center justify-center">
                <div className="text-center relative group">
                    <SubmitFormButtonComponent text="Log in" isLoading={false} />
                </div>
            </div>
        </FormComponent>
    )
}

export default LoginForm