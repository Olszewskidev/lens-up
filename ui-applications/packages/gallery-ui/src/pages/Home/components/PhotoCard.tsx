import { PhotoItem } from "../../../types/GalleryApiTypes"

interface IPhotoCardProps {
    photoItem: PhotoItem
}

const PhotoCard = ({ photoItem }: IPhotoCardProps) => {
    return (
        <div className="container h-full w-full text-center">
            <img
                key={`img-${photoItem.id}`}
                src={photoItem.url}
                alt={photoItem.id}
                className="h-full w-full object-contain"
            />
            <div className="absolute bottom-20 right-28 bg-white py-6 px-6 rounded-3xl w-64 my-4 shadow-xl">
                <div className="text-white flex items-center absolute rounded-full py-4 px-4 shadow-xl bg-gray-300 left-4 -top-6">
                    <p className=" text-xl">
                        &#128204;
                    </p>
                </div>
                <div className="mt-8">
                    <p className="text-xl font-semibold my-2">&#10080; {photoItem.wishesText} &#10078;</p>
                    <div className="border-t-2"></div>
                    <p className="font-semibold text-base my-2 mb-2 float-right">{photoItem.authorName}</p>
                </div>
            </div>
        </div>

    )
}

export default PhotoCard