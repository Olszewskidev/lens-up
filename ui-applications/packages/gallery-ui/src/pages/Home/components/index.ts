import PhotoGallery from "./PhotoGallery";
import QRCodeCard from "./QRCodeCard";


export {
    PhotoGallery,
    QRCodeCard,
}