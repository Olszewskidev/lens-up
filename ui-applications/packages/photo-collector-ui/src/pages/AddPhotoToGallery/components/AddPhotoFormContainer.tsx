interface IAddPhotoFormContainerProps {
    children: React.ReactNode
}

const AddPhotoFormContainer = ({ children }: IAddPhotoFormContainerProps) => {
    return (
        <div className="container w-full mx-auto items-center py-32">
            <div className="max-w-sm mx-auto overflow-hidden items-center">
                {
                    children
                }
            </div>
        </div>
    )
}

export default AddPhotoFormContainer
