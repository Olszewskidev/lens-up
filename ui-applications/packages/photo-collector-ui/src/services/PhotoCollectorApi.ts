import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'

const baseQuery = fetchBaseQuery({
    baseUrl: import.meta.env.VITE_PHOTO_COLLECTOR_SERVICE_URL,
});

const temp = import.meta.env.VITE_PHOTO_COLLECTOR_SERVICE_URL;

export const photoCollectorApi = createApi({
    reducerPath: 'photoCollectorApi',
    baseQuery: baseQuery,
    endpoints: (builder) => ({
        addPhotoToGallery: builder.mutation<void, { enterCode: string, formData: FormData }>({
            query: (payload) => ({
                method: 'POST',
                url: `upload-photo/${payload.enterCode}`,
                body: payload.formData,
                formData: true
            }),
        }),
    }),
})

export const { useAddPhotoToGalleryMutation } = photoCollectorApi