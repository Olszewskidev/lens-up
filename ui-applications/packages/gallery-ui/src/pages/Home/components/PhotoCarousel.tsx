import { PhotoItem } from "../../../types/GalleryApiTypes"

import 'swiper/css';
import 'swiper/css/effect-coverflow';

import { Swiper, SwiperSlide } from 'swiper/react';
import { Autoplay, EffectCoverflow } from 'swiper/modules';
import { carouselOptions } from "../utils/constants";
import PhotoCard from "./PhotoCard";

interface IPhotoGalleryProps {
    photoItems: PhotoItem[]
}

const PhotoCarousel = ({ photoItems }: IPhotoGalleryProps) => {
    return (
        <Swiper
            slidesPerView={1}
            loop={true}
            autoplay={{
                delay: carouselOptions.delay,
                disableOnInteraction: false,
            }}
            speed={carouselOptions.speed}
            modules={[Autoplay, EffectCoverflow]}
            coverflowEffect={{
                rotate: 0,
                stretch: 0,
                depth: 100,
                modifier: 2.5,
            }}
            className="h-4/5 w-4/5 object-contain items-center justify-center" 
        >
            {photoItems.map((image) => (
                <SwiperSlide key={image.id}>
                    <PhotoCard photoItem={image} />
                </SwiperSlide>))}
        </Swiper>
    )
}

export default PhotoCarousel