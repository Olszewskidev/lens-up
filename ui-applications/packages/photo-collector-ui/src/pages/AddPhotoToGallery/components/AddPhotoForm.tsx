import { ChangeEvent } from "react";
import AddPhotoFormContainer from "./AddPhotoFormContainer";
import { FormComponent, SubmitFormButtonComponent } from "@lens-up/shared-components";

interface IAddPhotoFormProps {
    handleFormSubmit: () => void,
    handlePhotoInputChange: (event: ChangeEvent<HTMLInputElement>) => void,
    isLoading: boolean
    photo?: File
}

const AddPhotoForm = ({ handleFormSubmit, handlePhotoInputChange, isLoading, photo }: IAddPhotoFormProps) => {
    return (
        <AddPhotoFormContainer>
            <FormComponent handleFormSubmit={handleFormSubmit}>
                <div className="px-4 py-6">
                    <div className="max-w-sm p-6 mb-4 items-center mx-auto text-center cursor-pointer ">
                        {!photo && (
                            <>
                                <label htmlFor="dropzone-file" className="flex flex-col items-center justify-center w-full">
                                    <div className="flex flex-col items-center justify-center text-white">
                                        <svg className="w-8 h-8 mb-4" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#ffffff" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                                            <rect x="3" y="3" width="18" height="18" rx="2" /><circle cx="8.5" cy="8.5" r="1.5" /><path d="M20.4 14.5L16 10 4 20" />
                                        </svg>
                                        <p className="mb-2 text-sm"><span className="font-semibold">Click to upload your photo</span></p>
                                    </div>
                                    <input id="dropzone-file" type="file" className="hidden" onChange={handlePhotoInputChange} />
                                </label>
                            </>
                        )
                        }
                        {
                            photo && (
                                <div className="bg-white p-4">
                                    <img className="w-full h-auto" src={URL.createObjectURL(photo)} />
                                </div>
                            )
                        }

                    </div>
                    <div className="flex items-center justify-center">
                        <div className="text-center relative group">
                            <SubmitFormButtonComponent text="Upload" isLoading={isLoading} />
                        </div>
                    </div>
                </div>
            </FormComponent>
        </AddPhotoFormContainer>
    )
}

export default AddPhotoForm
