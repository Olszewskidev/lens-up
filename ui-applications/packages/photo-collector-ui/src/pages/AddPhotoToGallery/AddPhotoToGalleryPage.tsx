import { ChangeEvent, useState } from "react";
import { useAddPhotoToGalleryMutation } from "../../services/PhotoCollectorApi";
import { AddPhotoForm, AddPhotoFormSkeleton } from "./components"
import { ErrorCardComponent, SuccessAlertComponent } from "@lens-up/shared-components";
import { useParams } from "react-router-dom";

const AddPhotoToGalleryPage = () => {
    const { enterCode } = useParams();
    const [addPhoto, { isError, isSuccess, isUninitialized, isLoading }] = useAddPhotoToGalleryMutation();
    const [photo, setPhoto] = useState<File>();

    const handlePhotoInputChange = (event: ChangeEvent<HTMLInputElement>) => {
        // TODO: Validation
        const fileList = event.target.files
        const isEmpty = !fileList || fileList.length === 0;
        if (isEmpty) {
            return
        }
        const photo = fileList[0];
        setPhoto(photo)
    };

    const handleFormSubmit = () => {
        // TODO: Validation
        if (!photo || !enterCode) {
            return
        }
        const formData = new FormData();
        formData.append('File', photo);
        addPhoto({ enterCode, formData })
    }

    return (
        <>
            {
                isUninitialized && <AddPhotoForm
                    handleFormSubmit={handleFormSubmit}
                    handlePhotoInputChange={handlePhotoInputChange}
                    photo={photo}
                    isLoading={isLoading} />
            }
            {
                isError && <ErrorCardComponent />
            }
            {
                isSuccess && <SuccessAlertComponent />
            }
        </>
    )
}

export default AddPhotoToGalleryPage
