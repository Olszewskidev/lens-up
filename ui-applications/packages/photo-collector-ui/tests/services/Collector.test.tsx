import { waitFor, cleanup, fireEvent, getByTestId, render, screen } from '@testing-library/react';
import { BrowserRouter, Navigate, Route, Routes, useNavigate } from 'react-router-dom';
import { Provider } from 'react-redux';
import { expect, test, vi, beforeAll, afterEach, afterAll } from 'vitest';
import { photoCollectorApi, useAddPhotoToGalleryMutation } from '../../src/services/PhotoCollectorApi';
import { createAsyncThunk } from '@reduxjs/toolkit';
import { HttpResponse, http } from 'msw';
import { setupServer } from 'msw/node';
import AddPhotoToGalleryPage from '../../src/pages/AddPhotoToGallery/AddPhotoToGalleryPage';
import { store } from '../../src/app/store';
import { useParams } from "react-router-dom";
import { AddPhotoForm } from '../../src/pages/AddPhotoToGallery/components';
import { SuccessCardComponent } from '@lens-up/shared-components';

const photo = new File(["happy"], "../happy.png", { type: "image/png" });

const handlers = [
    http.post(`${import.meta.env.VITE_PHOTO_COLLECTOR_SERVICE_URL}/upload-photo/0`, async () => {
        return new Response(null, {
            status: 200,
            headers: {
                Allow: 'GET,HEAD,POST',
            },
        })
    })
]

const successCardRender = render(
    <SuccessCardComponent />
).asFragment();

const AddPhotoComponent = () => {
    const { enterCode } = useParams();
    const [addPhoto, { isError, isSuccess, isUninitialized, isLoading }] = useAddPhotoToGalleryMutation();

    const handleAddPhotoTest = () => {
        if (!enterCode)
            return;
        const formData = new FormData();
        formData.append('photo', photo)
        formData.append('AuthorName', "Author")
        formData.append('WishesText', "Happy")
        addPhoto({ enterCode, formData })
    }

    return (
        <>
            {
                isUninitialized && <div data-testid="test-component" onClick={(e) => { e.preventDefault(); handleAddPhotoTest() }} />
            }
            {
                isSuccess && <SuccessCardComponent />
            }
        </>
    )
}

/**
* @vitest-environment jsdom
*/
test("Add photo muttation must return successful photo addition api request", async () => {
    const server = setupServer(...handlers); server.listen();
    URL.createObjectURL = vi.fn();

    const { getByTestId, asFragment } = render(
        <Provider store={store}>
            <BrowserRouter>
                <Routes>
                    <Route path="/upload-photo/:enterCode" element={<AddPhotoComponent />} />
                </Routes>
                <Navigate to="/upload-photo/0" replace={true} />
            </BrowserRouter>
        </Provider>
    );

    fireEvent.click(getByTestId("test-component"));

    let successRender = asFragment();

    await waitFor(() => {
        successRender = asFragment();
        expect(successRender).toEqual(successCardRender);
    }).then(() => expect(successRender).toMatchSnapshot());
});