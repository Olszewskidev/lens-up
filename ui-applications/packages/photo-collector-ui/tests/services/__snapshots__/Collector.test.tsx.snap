// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add photo muttation must render success card 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden items-center"
  >
    <div
      class="px-16 py-14"
    >
      <div
        class="flex justify-center"
      >
        <div
          class="rounded-full p-6 bg-green-200"
        >
          <div
            class="flex h-16 w-16 items-center justify-center rounded-full p-4 bg-green-500"
          >
            <svg
              class="h-8 w-8 text-white"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.5 12.75l6 6 9-13.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <h3
        class="my-4 text-center text-3xl font-semibold text-gray-700"
      >
        Success!
      </h3>
      <p
        class="w-[230px] text-center font-normal text-gray-600"
      >
        Everything is ok.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Add photo muttation must return successful photo addition api request 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden items-center"
  >
    <div
      class="px-16 py-14"
    >
      <div
        class="flex justify-center"
      >
        <div
          class="rounded-full p-6 bg-green-200"
        >
          <div
            class="flex h-16 w-16 items-center justify-center rounded-full p-4 bg-green-500"
          >
            <svg
              class="h-8 w-8 text-white"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.5 12.75l6 6 9-13.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <h3
        class="my-4 text-center text-3xl font-semibold text-gray-700"
      >
        Success!
      </h3>
      <p
        class="w-[230px] text-center font-normal text-gray-600"
      >
        Everything is ok.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
